src/mono/wasm/runtime/jiterpreter-trace-generator.ts

```diff
@@ -652,21 +652,104 @@ export function generate_wasm_body (
                 append_bailout(builder, ip, BailoutReason.Overflow); // could be underflow but awkward to tell
                 builder.endBlock();
                 break;
 
+            /*
+             *  The native conversion opcodes for these are not specified for nan/inf, and v8
+             *  chooses to throw, so we have to do some tricks to identify non-finite values
+             *  and substitute INTnn_MIN, like clang would.
+             *  This attempts to reproduce what clang does in -O3 with no special flags set:
+             *
+             *  f64 -> i64
+             *
+             *  block
+             *  local.get       0
+             *  f64.abs
+             *  f64.const       0x1p63
+             *  f64.lt
+             *  i32.eqz
+             *  br_if           0                               # 0: down to label0
+             *  local.get       0
+             *  i64.trunc_f64_s
+             *  return
+             *  end_block                               # label0:
+             *  i64.const       -9223372036854775808
+             *
+             *  f32 -> i32
+             *
+             *  block
+             *  local.get       0
+             *  f32.abs
+             *  f32.const       0x1p31
+             *  f32.lt
+             *  i32.eqz
+             *  br_if           0                               # 0: down to label3
+             *  local.get       0
+             *  i32.trunc_f32_s
+             *  return
+             *  end_block                               # label3:
+             *  i32.const       -2147483648
+             */
+            case MintOpcode.MINT_CONV_I4_R4:
+            case MintOpcode.MINT_CONV_I4_R8:
+            case MintOpcode.MINT_CONV_I8_R4:
+            case MintOpcode.MINT_CONV_I8_R8: {
+                const isF32 = (opcode === MintOpcode.MINT_CONV_I4_R4) ||
+                        (opcode === MintOpcode.MINT_CONV_I8_R4),
+                    isI64 = (opcode === MintOpcode.MINT_CONV_I8_R4) ||
+                        (opcode === MintOpcode.MINT_CONV_I8_R8),
+                    limit = isI64
+                        ? 9223372036854775807 // this will round up to 0x1p63
+                        : 2147483648, // this is 0x1p31 exactly
+                    tempLocal = isF32 ? "temp_f32" : "temp_f64";
+
+                // Pre-load locals for the result store at the end
+                builder.local("pLocals");
+
+                // Load src
+                append_ldloc(builder, getArgU16(ip, 2), isF32 ? WasmOpcode.f32_load : WasmOpcode.f64_load);
+                builder.local(tempLocal, WasmOpcode.tee_local);
+
+                // Detect whether the value is within the representable range for the target type
+                builder.appendU8(isF32 ? WasmOpcode.f32_abs : WasmOpcode.f64_abs);
+                builder.appendU8(isF32 ? WasmOpcode.f32_const : WasmOpcode.f64_const);
+                if (isF32)
+                    builder.appendF32(limit);
```

Comment:
It feels wrong for float32 -> int64 to use an f32 limit value, but that's what clang seems to do.